commit 3c1526259ff39220001222661d2c8191ddf7d249
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Tue Oct 23 15:16:35 2018 +0100

    ASoC: googlevoicehat-codec: Use correct device when grabbing GPIO
    
    The fixup for the VoiceHAT in 4.18 incorrectly tried to find the
    sdmode GPIO pin under the card device, not the codec device.
    This failed, and therefore caused the device probe to fail.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/sound/soc/bcm/googlevoicehat-codec.c b/sound/soc/bcm/googlevoicehat-codec.c
index 1cbd3ed6899b..187706ce5f1e 100644
--- a/sound/soc/bcm/googlevoicehat-codec.c
+++ b/sound/soc/bcm/googlevoicehat-codec.c
@@ -50,9 +50,9 @@ static void voicehat_enable_sdmode_work(struct work_struct *work) {
 static int voicehat_component_probe(struct snd_soc_component *component) {
   struct voicehat_priv *voicehat = snd_soc_component_get_drvdata(component);
 
-  voicehat->sdmode_gpio = devm_gpiod_get(component->card->dev, "sdmode", GPIOD_OUT_LOW);
+  voicehat->sdmode_gpio = devm_gpiod_get(component->dev, "sdmode", GPIOD_OUT_LOW);
   if (IS_ERR(voicehat->sdmode_gpio)) {
-    dev_err(component->card->dev, "Unable to allocate GPIO pin\n");
+    dev_err(component->dev, "Unable to allocate GPIO pin\n");
     return PTR_ERR(voicehat->sdmode_gpio);
   }
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1214_3c1526259ff39220001222661d2c8191ddf7d249.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1214_3c1526259ff39220001222661d2c8191ddf7d249.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
